src/main/java/ai/lucidtech/las/sdk/Credentials.java

```diff
@@ -26,45 +45,123 @@ public Credentials() {
+    }
+
+    private Map readFromEnviron() {
+        Map<String, String> pairs = new HashMap<String, String>();
+        pairs.put("clientId", "LAS_CLIENT_ID");
+        pairs.put("clientSecret", "LAS_CLIENT_SECRET");
+        pairs.put("apiKey", "LAS_API_KEY");
+        pairs.put("authEndpoint", "LAS_AUTH_ENDPOINT");
+        pairs.put("apiEndpoint", "LAS_API_ENDPOINT");
+
+        Map<String, String> result = new HashMap<String, String>();
+
+        for (Map.Entry<String, String> entry : pairs.entrySet()) {
+            result.put(entry.getKey(), System.getenv(entry.getValue()));
+        }
+
+        return result;
     }
 
-    private void readCredentials(String credentialsPath) {
+    private void readFromFile(String credentialsPath) {
         // TODO Read INI files properly
         try(FileInputStream input = new FileInputStream(credentialsPath)) {
             Properties properties = new Properties();
             properties.load(input);
+            Map<String, String> credentialsFromEnviron = this.readFromEnviron();
 
-            this.accessKeyId = properties.getProperty("access_key_id");
-            this.secretAccessKey = properties.getProperty("secret_access_key");
+            this.clientId = properties.getProperty("client_id");
+            this.clientSecret = properties.getProperty("client_secret");
             this.apiKey = properties.getProperty("api_key");
+            this.apiEndpoint = properties.getProperty("api_endpoint");
+            this.authEndpoint = properties.getProperty("auth_endpoint");
         } catch (IOException ex) {
             ex.printStackTrace();
         }
     }
 
-    public String getAccessKeyId() {
-        return this.accessKeyId;
+    private JSONObject getClientCredentials(HttpClient httpClient) throws IOException {
+        HttpHost targetHost = new HttpHost("https://" + this.authEndpoint + "/oauth2/token?grant_type=client_credentials");
+
+        AuthCache authCache = new BasicAuthCache();
+        authCache.put(targetHost, new BasicScheme());
+
+        HttpUriRequest request = new HttpPost("https://" + this.authEndpoint + "/oauth2/token?grant_type=client_credentials");
+        request.addHeader("Content-Type", "application/x-www-form-urlencoded");
+        request.addHeader("Accept", "application/json");
+
+        String authString = this.clientId + ":" + this.clientSecret;
+        String encodedAuth = Base64.getEncoder().encodeToString(authString.getBytes(StandardCharsets.UTF_8));
+        request.addHeader("Authorization", "Basic " + encodedAuth);
+
+        HttpResponse response = httpClient.execute(request);
+        HttpEntity responseEntity = response.getEntity();
+        int status = response.getStatusLine().getStatusCode();
+
+        if (status != 200) {
+            System.out.println("Request headers: ");
+            printHeaders(request.getAllHeaders());
+            System.out.println();
+            System.out.println("Response headers: ");
```

Comment:
Is there a logging library we can use instead of system print?

Reply:
I have removed most `println`s. Logging library sounds like a good idea, but I suggest it's done in another PR.